error: format not a string literal and no format arguments

**Describe the bug**
`Code/GraphMol/SubstructLibrary/CMakeFiles/SubstructLibrary.dir/PatternFactory.cpp` is not compiled because of `format-security` error:

```
[ 97%] Building CXX object Code/GraphMol/SubstructLibrary/CMakeFiles/SubstructLibrary.dir/PatternFactory.cpp.o
cd /home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/redhat-linux-build/Code/GraphMol/SubstructLibrary && /usr/lib64/ccache/g++ -DBOOST_DISABLE_THREADS -DBOOST_SERIALIZATION_DYN_LINK -DRDKIT_DYN_LINK -DRDKIT_SUBSTRUCTLIBRARY_BUILD -DRDK_64BIT_BUILD -DRDK_BUILD_MAEPARSER_SUPPORT -DRDK_HAS_EIGEN3 -DRDK_USE_BOOST_SERIALIZATION -DRDK_USE_STRICT_ROTOR_DEFINITION -DSubstructLibrary_EXPORTS -I/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/External -I/usr/include/inchi -I/usr/include/python3.10 -I/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code -I/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/External/boost-numeric-bindings -isystem /usr/include/eigen3 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mpopcnt -Wno-deprecated -Wno-unused-function -fno-strict-aliasing -Wall -Wextra -fPIC -O2 -g -DNDEBUG -fPIC -std=gnu++14 -MD -MT Code/GraphMol/SubstructLibrary/CMakeFiles/SubstructLibrary.dir/PatternFactory.cpp.o -MF CMakeFiles/SubstructLibrary.dir/PatternFactory.cpp.o.d -o CMakeFiles/SubstructLibrary.dir/PatternFactory.cpp.o -c /home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/GraphMol/SubstructLibrary/PatternFactory.cpp
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp: In function 'const char* makeMolJSON(CROMol)':
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:432:43: warning: unused parameter 'data' [-Wunused-parameter]
  432 | extern "C" const char *makeMolJSON(CROMol data) {
      |                                    ~~~~~~~^~~~
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp: In function 'void* parseMolJSON(char*, bool)':
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:449:38: warning: unused parameter 'data' [-Wunused-parameter]
  449 | extern "C" CROMol parseMolJSON(char *data, bool warnOnFail) {
      |                                ~~~~~~^~~~
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:449:49: warning: unused parameter 'warnOnFail' [-Wunused-parameter]
  449 | extern "C" CROMol parseMolJSON(char *data, bool warnOnFail) {
      |                                            ~~~~~^~~~~~~~~~
In file included from /usr/include/pgsql/server/postgres.h:47,
                 from /home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/rdkit.h:41,
                 from /home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:101:
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp: In function 'void* MolAdjustQueryProperties(CROMol, const char*)':
/usr/include/pgsql/server/utils/elog.h:215:40: error: format not a string literal and no format arguments [-Werror=format-security]
  215 |         ereport(elevel, errmsg_internal(__VA_ARGS__))
      |                         ~~~~~~~~~~~~~~~^~~~~~~~~~~~~
/usr/include/pgsql/server/utils/elog.h:128:25: note: in definition of macro 'ereport_domain'
  128 |                         __VA_ARGS__, errfinish(__FILE__, __LINE__, PG_FUNCNAME_MACRO); \
      |                         ^~~~~~~~~~~
/usr/include/pgsql/server/utils/elog.h:215:9: note: in expansion of macro 'ereport'
  215 |         ereport(elevel, errmsg_internal(__VA_ARGS__))
      |         ^~~~~~~
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:777:7: note: in expansion of macro 'elog'
  777 |       elog(ERROR, e.what());
      |       ^~~~
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp: In function 'void* makeAvalonBFP(CROMol, bool, unsigned int)':
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:1573:38: warning: unused parameter 'data' [-Wunused-parameter]
 1573 | extern "C" CBfp makeAvalonBFP(CROMol data, bool isQuery,
      |                               ~~~~~~~^~~~
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:1573:49: warning: unused parameter 'isQuery' [-Wunused-parameter]
 1573 | extern "C" CBfp makeAvalonBFP(CROMol data, bool isQuery,
      |                                            ~~~~~^~~~~~~
/home/sagitter/rpmbuild/BUILD/rdkit-Release_2022_03_1/Code/PgSQL/rdkit/adapter.cpp:1574:44: warning: unused parameter 'bitFlags' [-Wunused-parameter]
 1574 |                               unsigned int bitFlags) {
      |                               ~~~~~~~~~~~~~^~~~~~~~
cc1plus: some warnings being treated as errors
gmake[2]: *** [Code/PgSQL/rdkit/CMakeFiles/rdkit.dir/build.make:79: Code/PgSQL/rdkit/CMakeFiles/rdkit.dir/adapter.cpp.o] Error 1
```

**To Reproduce**
Compiling in Fedora 35 with this CMake command:

```
+ CFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection'
+ export CFLAGS
+ CXXFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection'
+ export CXXFLAGS
+ FFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules'
+ export FFLAGS
+ FCFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules'
+ export FCFLAGS
+ LDFLAGS='-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 '
+ export LDFLAGS
+ LT_SYS_LIBRARY_PATH=/usr/lib64:
+ export LT_SYS_LIBRARY_PATH
+ CC=gcc
+ export CC
+ CXX=g++
+ export CXX
++ /usr/bin/pg_server_config --includedir-server
+ /usr/bin/cmake -S . -B redhat-linux-build -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON -D RDK_INSTALL_INTREE=OFF -D CMAKE_BUILD_TYPE=RelWithDebInfo -D RDK_BUILD_INCHI_SUPPORT:BOOL=ON -D RDK_BUILD_THREADSAFE_SSS:BOOL=OFF -D RDK_BUILD_CAIRO_SUPPORT:BOOL=ON -D RDK_BUILD_DESCRIPTORS3D:BOOL=OFF -D RDK_BUILD_COORDGEN_SUPPORT:BOOL=OFF -D RDK_BUILD_MOLINTERCHANGE_SUPPORT:BOOL=OFF -D RDK_BUILD_PGSQL:BOOL=ON -D PostgreSQL_ROOT:PATH=/usr -D PostgreSQL_TYPE_INCLUDE_DIR:PATH=/usr/include/pgsql/server -D RDK_PGSQL_STATIC:BOOL=OFF -D RDK_INSTALL_STATIC_LIBS:BOOL=OFF -D RDK_USE_FLEXBISON:BOOL=OFF -D RDK_TEST_MULTITHREADED:BOOL=ON -D PYTHON_EXECUTABLE:FILEPATH=/usr/bin/python3 -D RDK_BOOST_PYTHON3_NAME=python310 -D CATCH_DIR:FILEPATH=/usr/include/catch2 -D RDK_USE_URF:BOOL=OFF -D RDK_INSTALL_COMIC_FONTS:BOOL=OFF -D RDK_BUILD_SWIG_JAVA_WRAPPER:BOOL=OFF -D RDK_BUILD_SWIG_WRAPPERS=OFF
```

**Configuration (please complete the following information):**
 - RDKit version: 2022.03.1
 - OS: Fedora 35
 - Python version (if relevant): 3.10
 - Are you using conda? No

> /usr/include/pgsql/server/utils/elog.h:215:40: error: format not a string literal and no format arguments [-Werror=format-security]

Probably, it's not an issue of RDKIT

--- a/Code/PgSQL/rdkit/adapter.cpp
+++ b/Code/PgSQL/rdkit/adapter.cpp
@@ -774,7 +774,7 @@ extern "C" CROMol MolAdjustQueryProperties(CROMol i, const char *params) {
     try {
       MolOps::parseAdjustQueryParametersFromJSON(p, pstring);
     } catch (const ValueErrorException &e) {
-      elog(ERROR, e.what());
+      elog(ERROR, "MolAdjustQueryProperties: %s", e.what());
     } catch (...) {
       elog(WARNING,
            "adjustQueryProperties: Invalid argument \'params\' ignored");

--- a/Code/PgSQL/rdkit/expected/rdkit-91.out
+++ b/Code/PgSQL/rdkit/expected/rdkit-91.out
@@ -1307,7 +1307,7 @@ select 'C1C(C)C1CCCC'::mol @> mol_adjust_query_properties('C1CC1CC'::mol,'{"adju
 (1 row)
 
 select 'C1C(C)C1CCCC'::mol @> mol_adjust_query_properties('C1CC1CC'::mol,'{"adjustDegreeFlags":"bogus"}');
-ERROR:  Unknown flag value: 'BOGUS'. Flags ignored.
+ERROR:  MolAdjustQueryProperties: Unknown flag value: 'BOGUS'. Flags ignored.
 select 'C1C([2H])C1CCCC'::mol @> mol_adjust_query_properties('C1CC1CC'::mol);
  ?column? 
 ----------

